nml/actions/action2layout.py

```diff
@@ -353,13 +353,7 @@ def _validate_bounding_box(self, name, value):
         return expression.ConstantNumeric(0)
 
     def _validate_hide_sprite(self, name, value):
-        value = expression.Not(value)
-        try:
-            value = value.reduce()
-        except:
-            pass
-        self.create_register(name, value)
-        return None
+        self.create_register(name, expression.Not(value).reduce())
```

Comment:
Are we sure `reduce()` can't throw ?

Reply:
It can raise a `ScriptError` if the expression doesn't return an integer. The only meaningful values for `hide_sprite` are 0 and 1, so in that case we should fail compilation.

(I did wonder why this try-except was added originally, but couldn't find an explanation)